### ISSUE TITLE
stuck with outline manager step 2

### ISSUE THREAD

**Issue by g3grau** [Jigsaw-Code/outline-server#1078]
I tried several installations both on AWS and on a local server using the command from step 1 using the "setup anywhere" button.
Installation asked to open two specific ports for TCP/UDP which has been done, I can connect with telnet to both ports...
However, pasting the {apiurl...} string into step 2, outline manager always repeats the message that the ports are likely not open and suggests to open anything from 1024 to 64k. That sounds like a bad idea. Does it really need more ports than the ones reported during the installation?
How to debug what fails? What is the outline manager trying to do at this point? How else can I get the key to connect from the outline client to bypass the manager? I assume that creation of the key is the only step missing, and with access to the server I would guess that I don't need the manager for that?

**Comment by bemasc** [comment]
Can you reach the API URL in a web browser?  It should connect and show a certificate warning.  If your browser instead shows a generic connection failure, then something else is wrong.

**Comment by g3grau** [comment]
Hi bemasc,

the API URL from the installation process is "somewhat" reachable but returns some kind of web page with a message {"code":"ResourceNotFound","message":"/xyzunreadable123 does not exist"} .. which was not very helpful. (and yes, after a certificate warning)

It took a while until I found relevant documentation snippets here: 
[https://github.com/Jigsaw-Code/outline-server/blob/master/src/shadowbox/README.md](https://github.com/Jigsaw-Code/outline-server/blob/master/src/shadowbox/README.md)
Using the approach:
```
API_URL=https://1.2.3.4:5678/xyz123abc
curl --insecure $API_URL/access-keys/
(taken from the string to be fed to the outline manager) I got a useful response in the form
{"accessKeys":[{"id":"0","name":"","password":"blah","port":12345,"method":"chacha20-ietf-poly1305","accessUrl":["ss://unreadable@serverIP:port/?outline=1"] ...
```
of which I could use the ss://...../?outline=1 part to connect to the server using the Outline-Client! :)

One mystery though: the local installation indeed used the two ports mentioned during the installation, but the installation on my AWS instance returned a 3rd port in this response ... and only after opening that for TCP+UDP I could connect.
In effect, the outline manager had reason to complain on one installation (for the 3rd port still being closed), but the other should have worked.

I'm clueless about the 3rd port. The 1st server installation on AWS failed (nano instance seems to be too small) and I reinstalled it .. could this cause a wrong port listing?

It was a bit scary to get a root password request on the 1st connection with the outline-client. I figured out that this installed a new interface outline-tun0 and changed the route so that the entire system communication goes over the outline server (that's the part that I couldn't figure out with the normal SS server). A first attempt to run NX over this connection was quite laggy, but more stable compared to a Qt-SS client with proxychains. 

Now I still seem to have some hickups to disable the tunnel. When I disconnect Outline the tunnel remains active and the internet connection is lost. It seems that I have to manually run ifconfig outline-tun0 down when running without outline and to re-enable it with ifconfig outline-tun0 up before reconnecting. Is that expected or should this be handled by the outline client? After closing the outline client, quite a few processes remain active such as OutlineProxyController and several outline-client instances. I guess I need to play a bit with it, including system restarts, to figure out how to use it on demand.
